### PR TITLE
test cases for JSON marshalling

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
Related Issue: #55

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@gmlewis All test case passes, still travis failed 😞. Please let me know the possible fix

**Comment by gmlewis** [comment]
Without looking at the Travis CI logs, I can see that at least the first file has not been run through `gofmt`.

Please see [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) and follow the directions there.  Thanks.

**Comment by vaibhavsingh97** [comment]
@gmlewis I already ran `go fmt ./...` on my patch

**Comment by gmlewis** [comment]
Looking at the logs from Travis CI, it says:

```
$ diff -u <(echo -n) <(gofmt -d -s .)
--- /dev/fd/63	2019-04-02 22:46:57.657608030 +0000
+++ /dev/fd/62	2019-04-02 22:46:57.657608030 +0000
@@ -0,0 +1,12 @@
+diff -u github/gists_test.go.orig github/gists_test.go
+--- github/gists_test.go.orig	2019-04-02 22:46:57.721579604 +0000
++++ github/gists_test.go	2019-04-02 22:46:57.721579604 +0000
+@@ -43,7 +43,7 @@
+ 			URL:         String("u"),
+ 		},
+ 		Files: map[GistFilename]GistFile{
+-			"gistfile.py": GistFile{
++			"gistfile.py": {
+ 				Size:     Int(167),
+ 				Filename: String("gistfile.py"),
+ 				Language: String("Python"),
The command "diff -u <(echo -n) <(gofmt -d -s .)" exited with 1.
```

What version of `go` are you using?
After running `go fmt ./...`, did you remember to commit the changes before pushing to your PR?
What does `git status` say in your repo?

**Comment by vaibhavsingh97** [comment]
go version go1.11.5

image

I did commit the changes, before sending the PR. Should I retrigger travis, might be there was some issue?

**Comment by gmlewis** [comment]
No, there is no need, as it is obviously not formatted properly.
Click on the `Files changed` tab in GitHub and scroll down... you will see misalignments in many places.
Try pushing to the PR again with the correctly-formatted code.

**Comment by gmlewis** [comment]
Here's an example:
Screenshot from 2019-04-03 07-34-51

**Comment by vaibhavsingh97** [comment]
Gotcha! Will update after proper formatting the code 👍

File: github/gists_test.go [hunk 1]
```diff
@@ -15,6 +15,215 @@ import (
...
+			Email:       String("e"),
+			Hireable:    Bool(true),
+			PublicRepos: Int(1),
+			Followers:   Int(1),
+			Following:   Int(1),
+			CreatedAt:   &Timestamp{referenceTime},
+			URL:         String("u"),
+		},
+		Files: map[GistFilename]GistFile{
+			"gistfile.py": GistFile{
```
Comment (vaibhavsingh97) [inline github/gists_test.go]: Found the issue, `go fmt` asking me to remove `GistFile` and use this instead 

```go
	"gistfile.py": {
         ...
```

**Comment by gmlewis** [comment]
Thank you, @gauntface!
Merging.

**Pull request merged**